### PR TITLE
feat: Add tax region + rates to SDK and types

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/core/types/src/http/tax-rate/admin/entities.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export interface AdminTaxRate {
+  id: string
+  rate: number | null
+  code: string | null
+  name: string
+  metadata: Record<string, unknown> | null
+  tax_region_id: string
+  is_combinable: boolean
+  is_default: boolean
+  created_at: string | Date
```
Comment (kasperkristensen) [inline packages/core/types/src/http/tax-rate/admin/entities.ts]: rm: `Date`

**Review by sradevski — APPROVED** [review]
Nice 👍

File: packages/core/js-sdk/src/admin/tax-rate.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        method: "POST",
+        headers,
+        body,
+        query,
+      }
+    )
+  }
+
+  async delete(id: string, headers?: ClientHeaders) {
+    return await this.client.fetch<any>(`${taxRateUrl}/${id}`, {
```
Comment (kasperkristensen) [inline packages/core/js-sdk/src/admin/tax-rate.ts]: should we add AdminTaxRateDeleteResponse?

File: packages/core/js-sdk/src/admin/tax-rate.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+export class TaxRate {
+  private client: Client
+  constructor(client: Client) {
+    this.client = client
+  }
+
+  async create(
+    body: HttpTypes.AdminCreateTaxRate,
+    query?: SelectParams,
```
Comment (kasperkristensen) [inline packages/core/js-sdk/src/admin/tax-rate.ts]: nit: We can just call `HttpTypes.SelectParams`

**Review by kasperkristensen — APPROVED** [review]
LGTM, just some minor things to address before merging

**Review by fPolic — APPROVED** [review]
LGTM

File: packages/core/types/src/http/tax-region/admin/queries.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { OperatorMap } from "../../../dal"
+import { FindParams } from "../../common"
+
+export interface AdminTaxRegionListParams extends FindParams {
+  id?: string | string[]
+  q?: string
+  parent_id?: string | string[] | OperatorMap<string>
+  country_code?: string | string[] | OperatorMap<string>
+  province_code?: string | string[] | OperatorMap<string>
```
Comment (adrien2p) [inline packages/core/types/src/http/tax-region/admin/queries.ts]:
```suggestion
  parent_id?: string | string[] | OperatorMap<string | string[]>
  country_code?: string | string[] | OperatorMap<string | string[]>
  province_code?: string | string[] | OperatorMap<string | string[]>
```

File: packages/core/types/src/http/tax-region/admin/queries.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  created_at?: OperatorMap<string>
+  updated_at?: OperatorMap<string>
+  deleted_at?: OperatorMap<string>
+  created_by?: OperatorMap<string>
```
Comment (adrien2p) [inline packages/core/types/src/http/tax-region/admin/queries.ts]:
```suggestion
  created_at?: string | OperatorMap<string>
  updated_at?: string | OperatorMap<string>
  deleted_at?: string | OperatorMap<string>
  created_by?: string | OperatorMap<string>
```

File: packages/core/types/src/http/tax-rate/admin/queries.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { OperatorMap } from "../../../dal"
+import { FindParams } from "../../common"
+
+export interface AdminTaxRateListParams extends FindParams {
+  id?: string | string[]
+  q?: string
+  tax_region_id?: string | string[] | OperatorMap<string>
```
Comment (adrien2p) [inline packages/core/types/src/http/tax-rate/admin/queries.ts]:
```suggestion
  tax_region_id?: string | string[] | OperatorMap<string | string[]>
```

File: packages/core/types/src/http/tax-rate/admin/queries.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  is_default?: string
+  service_zone_id?: string
+  shipping_profile_id?: string
+  provider_id?: string
+  shipping_option_type_id?: string
+  created_at?: OperatorMap<string>
+  updated_at?: OperatorMap<string>
+  deleted_at?: OperatorMap<string>
```
Comment (adrien2p) [inline packages/core/types/src/http/tax-rate/admin/queries.ts]:
```suggestion
  created_at?: string | OperatorMap<string>
  updated_at?: string | OperatorMap<string>
  deleted_at?: string | OperatorMap<string>
```

**Review by adrien2p — APPROVED** [review]
LGTM

File: packages/core/types/src/http/tax-rate/admin/entities.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/core/types/src/http/tax-rate/admin/entities.ts]: Good catch

**Pull request merged**